components/logging/org.wso2.carbon.logging.appender.http/src/main/java/org/wso2/carbon/logging/appender/http/SecuredHttpAppender.java

```diff
@@ -356,21 +399,80 @@ private String resolveSecretPassword(String password) {
+                    this.failureWarningLevel = FailureWaringLevel.OVER_90_PERCENT;
+                }
+                break;
+            case OVER_90_PERCENT:
+                if (diskUsage >= persistentQueue.getMaxDiskSpaceInBytes()) {
+                    getStatusLogger().warn("Remote log publishing failure : The number of logs in the queue has exceeded" +
+                            " the allocated queue limit. Please check the remote server status.");
+                    this.failureWarningLevel = FailureWaringLevel.FULLY_USED;
+                }
+                break;
+            case FULLY_USED:
+                getStatusLogger().warn("Remote log publishing failure : Allocated queue limit reached. Starting to loose" +
+                        " logs. Please check the remote server status.");
+                this.failureWarningLevel = FailureWaringLevel.LOOSING_LOGS;
+                this.lastLogLossWarningTime = new Date().getTime();
+                break;
+            case LOOSING_LOGS:
+                long timeSinceLastWarning = new Date().getTime() - lastLogLossWarningTime;
+                int FAILURE_WARNING_DELAY_MINUTES = 1;
+                if (timeSinceLastWarning > TimeUnit.MINUTES.toMillis(FAILURE_WARNING_DELAY_MINUTES)) {
+                    getStatusLogger().warn("Remote log publishing failure : Allocated queue limit reached. Unsaved" +
+                            "logs are being dumped. Please check the remote server status.");
+                    this.lastLogLossWarningTime = new Date().getTime();
+                }
+                break;
+        }
+    }
+
+    private void resetWarningLogOnRemoteServerSuccess() {
+
+        if(this.failureWarningLevel != FailureWaringLevel.NONE) {
+            getStatusLogger().info("Remote log publishing success : Remote server is up and running.");
+            this.failureWarningLevel = FailureWaringLevel.NONE;
+        }
+    }
+
     private final class LogPublisherTask implements Runnable {
+
         @Override
         public void run() {
-            // print all the messages in the queue to the console if the queue is not empty
-            if (isManagerInitialized && !queue.isEmpty()) {
+
+            if (isManagerInitialized && !persistentQueue.isEmpty()) {
+                // publish logs from the queue
                 try {
-                    LogEvent event = queue.take();
-                    manager.send(getLayout(), event);
-                } catch (InterruptedException e) {
-                    Thread.currentThread().interrupt();
-                    error("Error occurred while publishing logs to HTTP endpoint", e);
+                    LogEvent event = persistentQueue.peek();
+                    if (event != null) {
+                        manager.send(getLayout(), event);
+                        persistentQueue.dequeue();
+                        resetWarningLogOnRemoteServerSuccess();
```

Comment:
here check the current level and update

Reply:
Checking the current level is moved to where the initial log is printed. 
Reason : The initial log to inform that the remove server is failing is printed at the start of each failure period. Then the current level is calculated and the relevant warnings are printed from there on.